bodhi/server/models.py

```diff
@@ -2543,8 +2543,12 @@ def test_cases(self):
     def full_test_cases(self):
         tests = set()
         for build in self.builds:
-            for test in build.package.test_cases:
-                tests.add(test)
+            test_cases = build.package.test_cases
+            unique_tests = dict()
+            for test in test_cases:
+                if not unique_tests.get(test.name):
+                    unique_tests[test.name] = True
+                    tests.add(test)
```

Comment:
So I think you could rework the above to be:

```
test_names = set()
for test in build.package.test_cases:
    if test.name not in test_names:
        test_names.add(test.name)
        tests.add(test)
```

In fact, it's a little curious that the ```tests``` set doesn't "uniquify" this already.